### ISSUE TITLE
Combining forces

### ISSUE THREAD

**Issue by rwwagner90** [icicletech/ember-cli-full-calendar#52]
Hello,

We have been working on another ember fullcalendar addon, and we thought it might be good to try to merge them together and work on one gold standard, rather than working separately. What do you guys think?

https://github.com/scoutforpets/ember-fullcalendar

**Comment by DanChadwick** [comment]
@rwwagner90: As a consummer, I would love to see one great module than two good modules. It might help to understand the differences between the two modules.

**Comment by rwwagner90** [comment]
@DanChadwick I think the fundamental difference is that our addon allows for passing any fullcalendar option without having to create a new placeholder function or property in the addon. You can just add it to the array of possible options, and it passes it to fullcalendar. 

Another thing that we have is the fullcalendar-scheduler.

@jamesdixon could speak more to the advantages of our implementation, as he wrote it.

**Comment by DanChadwick** [comment]
Other than the forced bundling with fc-scheduler, what are the disadvantages?

**Comment by shrivara** [comment]
@rwwagner90 Absolutely, will be great if we can merge. How do you want to go about it?

**Comment by RobbieTheWagner** [comment]
@shrivara I'm not sure. I'm not very familiar with fullcalendar in the first place. @jamesdixon would you be able to articulate some of the features our addon gives that we definitely want to keep?

**Comment by jamesdixon** [comment]
Sorry for the late reply, everyone.

@DanChadwick: @rwwagner90 is correct -- those are the two main differences. The ability to simply add new properties, events, etc to the options array makes it a much easier to maintain IMO. The other addition is the inclusion of FullCalendar scheduler, which we're using. That said, I had made a note to make that optional and lighten the payload if you're not using it, but hadn't gotten around to it.

As for disadvantages, I'm not sure there are really any other than the bundling of FCS if you're not using it. @shrivara's implementation does have a nice feature to be able to send actions down to FC to invoke methods, such as `prev()`. I'd like to get something like that setup as well.

@shrivara in terms of combining forces, the one major thing we'd need to agree on is which implementation to start from and then work from there. I may be biased, but I personally think that my approach is more maintainable going forward and would require less work as new methods/properties/etc. are added to FC and FC Scheduler. I'd be interested to hear everyone else's opinion.

Here's my list of items to get done eventually:
1. Implement a system for calling FC methods, such as `prev()` -- similar to what you have now.
2. Make FC Scheduler optional
3. Dig more into what can be done the "ember way" (if it makes sense). For example, if you check my repo, there a few pulls still outstanding for additions to allow for interacting with the calendar from outside the component. It would be good to do a more holistic investigation to determine what does and does not make sense and the best way to implement it.
4. More tests. As of now, my repo only has a couple that a contributor wrote. If things are going to be written to be more idiomatic ember, then more tests should be written.

Would love to hear everyone's thoughts.

**Comment by ghost** [comment]
@jamesdixon Thanks for the write up and really sorry about the late reply. Unfortunately, I'm not actively maintaining this anymore, so this might take a while to get going.
Pinging @wicliff @prasannatm to get this going if possible.

**Comment by jamesdixon** [comment]
No worries, @shrivaraks!

**Comment by DanChadwick** [comment]
`ember-fullcalendar` how has scheduler as optional, as well as DDAU for date and viewName. I would suggest abandoning `ember-cli-full-calendar` and directing all development to `ember-fullcalendar`. 

Better to all pull together on one module, I think. I would suggest one final commit to update the readme to recomment ember-fullcalendar. When I migrated, the only changes I need were for the two DDAU properties mentioned above.

**Comment by RobbieTheWagner** [comment]
@wicliff @prasannatm @shrivaraks @jamesdixon @DanChadwick can we all agree to work on ember-fullcalendar and mark this as deprecated?

**Comment by jamesdixon** [comment]
@rwwagner90 fine by me. That said, I believe I'm going to be looking for maintainers and shifting my development efforts towards the [ember-calendar](https://github.com/alphasights/ember-calendar) project. While not as full-featured as Fullcalendar, it's a pure Ember implementation and not only seemingly more performant, but much easier to work with.

I've been working with @marzubov on [his fork](https://github.com/marzubov/ember-calendar), which includes the much-needed day and month view that's currently missing in the main repo. In addition, we've stripped out unused and unnecessary dependencies in an effort to make the package much more lean. Would love to have everyone's help on making it even better!

All that said, if anyone would be interested in being a maintainer on the ember-fullcalendar project, please let me know.

**Comment by Blackening999** [comment]
@jamesdixon it's sad. ember-calendar is a very weak in comparison with fullcalendar which is full-featured and production ready :(

**Comment by jamesdixon** [comment]
@Blackening999 every piece of software has to start somewhere. Sure, FullCalendar has a lot of features, but it's not without its faults. In particular, rendering large numbers of events is extremely slow and is a known issue that has yet to be resolved. I prefer ember-calendar's "native" approach and working with everything as components.

**Comment by Blackening999** [comment]
@jamesdixon I agreed with you. The issue is that you're saying that you're not going to support something is really useful and this is sad :( I like ember-calendar in its core but it lacks at least so simple features like "weekly/monthly" view and this makes it almost useless for most of the business needs. If I'd have a time I'd contribute there. For now, full calendar seems to be the only serious player on the ember-addons market

**Comment by jamesdixon** [comment]
@Blackening999 I'm saying that I'm shifting the majority of my development efforts to ember-calendar. I certainly will continue to support my FullCalendar addon until I have maintainers.

Also, as I mentioned above, the fork I'm working on contains month and day view, which I agree is a must.
